packages/shell-api/src/collection.ts

```diff
@@ -2115,12 +2139,35 @@ export default class Collection extends ShellApiWithMongoClass {
       avgObjSize: number;
     }[] = [];
 
+    const ns = `${this._database._name}.${this._name}`;
+    let configCollectionsInfo: Document;
+    const existingConfigCollectionsInfo = await config
+      .getCollection('collections')
+      .findOne({
+        _id: ns,
+        ...onlyShardedCollectionsInConfigFilter,
+      });
+
+    if (!existingConfigCollectionsInfo) {
+      const timeseriesCollectionInfo =
+        await this._getShardedTimeseriesCollectionInfo(config, collStats);
+
+      if (!timeseriesCollectionInfo) {
+        throw new MongoshInvalidInputError(
+          `Collection ${this._name} is not sharded`,
+          ShellApiErrors.NotConnectedToShardedCluster
+        );
+      }
+
+      configCollectionsInfo = timeseriesCollectionInfo;
+    } else {
+      configCollectionsInfo = existingConfigCollectionsInfo;
+    }
```

Comment:
Just a feeling, but my gut reaction here is that it would make sense to move this conditional into a helper like `_getShardedTimeseriesCollectionInfo()` as well – if we end up re-using `_getShardedTimeseriesCollectionInfo()` at some point, there's a good chance that we'll want the rest of this block as well too